src/components/ContextMenuView/ChatContextMenuView.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import ContextMenu, {
+  ContextMenuAction,
+  ContextMenuOnPressNativeEvent
+} from 'react-native-context-menu-view';
+import {NativeSyntheticEvent} from 'react-native';
+
+import ShareUtils from '../../utils/share';
+import {COLORS} from '../../utils/theme';
+
+type BetterSocialContextMenuType = 'MyChatContextMenu' | 'TargetChatContextMenu';
+type BetterSocialContextMenuName = 'Copy' | 'Delete' | 'Reply';
+type BetterSocialContextMenuAction = ContextMenuAction & {title: BetterSocialContextMenuName};
+type BetterSocialContextMenuOnPressNativeEvent = ContextMenuOnPressNativeEvent & {
+  name: BetterSocialContextMenuName;
+};
+
+type ChatContextMenuViewProps = {
+  children: React.ReactNode;
+  message: string;
+  contextMenuType: BetterSocialContextMenuType;
+};
+
+const MyChatContextMenu: BetterSocialContextMenuAction[] = [
+  {title: 'Copy', systemIcon: 'square.on.square'}
+  //   {title: 'Delete', systemIcon: 'trash', destructive: true}
+];
+
+const TargetChatContextMenu: BetterSocialContextMenuAction[] = [
+  {title: 'Copy', systemIcon: 'square.on.square'}
+];
+
+const ChatContextMenuView = ({children, message, contextMenuType}: ChatContextMenuViewProps) => {
+  let contextMenuActions = MyChatContextMenu;
+  if (contextMenuType === 'TargetChatContextMenu') {
+    contextMenuActions = TargetChatContextMenu;
+  }
+
+  const onContextMenuPressed = (e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>) => {
+    console.log('context menu pressed', message);
+    const event = e?.nativeEvent as BetterSocialContextMenuOnPressNativeEvent;
+
+    if (event.name === 'Copy') {
+      ShareUtils.copyMessageWithoutLink(message);
+    }
+  };
+
+  return (
+    <ContextMenu
+      previewBackgroundColor={COLORS.transparent}
+      style={{flex: 1}}
+      actions={contextMenuActions}
+      onPress={onContextMenuPressed}>
+      {children}
+    </ContextMenu>
+  );
+};
+
+export default ChatContextMenuView;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a console.log statement on line 40 which should be removed. Console logs are not recommended for production code as they can expose sensitive information and slow down the application.

```diff
-    console.log('context menu pressed', message);
```

Also, it seems like you have commented out the 'Delete' action in `MyChatContextMenu` (line 26). If this is intentional and you plan to implement it later, consider adding a TODO comment to clarify this. If it's not needed, it's better to remove the commented code to keep the codebase clean.

```diff
-   //   {title: 'Delete', systemIcon: 'trash', destructive: true}
+   // TODO: Implement 'Delete' action
```

Lastly, I noticed that you're using a type check to determine which context menu actions to use (lines 34-37). This could be simplified by using a mapping object, reducing the need for conditional logic and making the code more maintainable if more types are added in the future.

```diff
- let contextMenuActions = MyChatContextMenu;
- if (contextMenuType === 'TargetChatContextMenu') {
-   contextMenuActions = TargetChatContextMenu;
- }

+ const contextMenuActionsMap = {
+   MyChatContextMenu,
+   TargetChatContextMenu
+ };
+ const contextMenuActions = contextMenuActionsMap[contextMenuType];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->